### PR TITLE
Fix requireReadOnlyReactProps

### PR DESCRIPTION
**Pull request by kangax**
- `$FlowFixMe` is ignored
- `{||}` is considered to be read-only

### CODE REVIEW

File: src/rules/requireReadonlyReactProps.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const schema = [];
 
 const reComponentName = /^(Pure)?Component$/;
+const reReadOnly = /^\$(ReadOnly|FlowFixMe)$/;
```
Comment (mrtnzlml) [inline src/rules/requireReadonlyReactProps.js]: `$FlowFixMe` was just an example. It can be anything defined in `suppress_type`.

Comment (kangax) [reply src/rules/requireReadonlyReactProps.js]: how would we know what's defined there (from within a plugin rule)?

Comment (mrtnzlml) [reply src/rules/requireReadonlyReactProps.js]: You don't. This is why I was suggesting to not throw an error in cases where you don't know it's wrong for sure. :)

Comment (pnevyk) [reply src/rules/requireReadonlyReactProps.js]: Maybe allow to specify the suppress type name in an option of the rule?

**Pull request merged**